File: src/api-gateway/ApiGateway/Models/Post.cs
namespace ApiGateway.Models;

public class Post
{
    public required Ulid Id { get; init; }

    public required UserSummary User { get; init; }
    
    public required string Text { get; init; }
    
    public required DateTime CreatedAt { get; init; }

    public required DateTime UpdatedAt { get; init; }

    public IEnumerable<FileMetadata>? FilesMetadata { get; init; }

    public IEnumerable<Reply>? Replies { get; set; }
    
    public bool? UserInteracted { get; set; }

    public int Likes { get; set; }

    public int Views { get; set; }
}

File: src/web-ui/src/models/Post.ts
﻿import {FileMetadata} from "@/models/FileMetadata.ts";
import {Reply} from "@/models/Reply.ts";

export interface Post {
    id: string;
    content: string;
    likes: number;
    views: number;
    user: UserPost;
    filesMetadata: FileMetadata[] | null;
    replies: Reply[] | null;
    createdAt: string;
    updatedAt: string;
    userInteracted: boolean;
}

export interface UserPost {
    id: string;
    profileImage: FileMetadata;
    name: string;
}

